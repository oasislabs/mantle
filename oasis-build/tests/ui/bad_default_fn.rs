use oasis_std::{Context, Service};

#[derive(Service)]
pub struct MyService;

impl MyService {
    pub fn new(_ctx: &Context) -> Self {
        Self
    }

    #[oasis_std::default]
    pub fn bad_default(&mut self, _ctx: &Context, arg: u8) {}
}

fn main() {
    oasis_std::service!(MyService);
}
