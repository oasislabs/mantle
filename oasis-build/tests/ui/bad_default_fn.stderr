error: Default function cannot take arguments after `Context`.
  --> $DIR/bad_default_fn.rs:12:51
   |
12 |     pub fn bad_default(&mut self, _ctx: &Context, arg: u8) {}
   |                                                   ^^^^^^^

error: aborting due to previous error

